addons/account/models/account_invoice.py

```diff
@@ -946,39 +920,8 @@ def _notify_get_groups(self, message, groups):
 
         return groups
 
-    @api.multi
-    def get_access_action(self, access_uid=None):
-        """ Instead of the classic form view, redirect to the online invoice for portal users. """
-        self.ensure_one()
-        user, record = self.env.user, self
-        if access_uid:
-            user = self.env['res.users'].sudo().browse(access_uid)
-            record = self.sudo(user)
-
-        if user.share or self.env.context.get('force_website'):
-            try:
-                record.check_access_rule('read')
-            except exceptions.AccessError:
-                if self.env.context.get('force_website'):
-                    return {
-                        'type': 'ir.actions.act_url',
-                        'url': '/my/invoices/%s' % self.id,
-                        'target': 'self',
-                        'res_id': self.id,
-                    }
-                else:
-                    pass
-            else:
-                return {
-                    'type': 'ir.actions.act_url',
-                    'url': '/my/invoices/%s?access_token=%s' % (self.id, self.access_token),
-                    'target': 'self',
-                    'res_id': self.id,
-                }
-        return super(AccountInvoice, self).get_access_action(access_uid)
-
-    def get_mail_url(self):
-        return self.get_share_url()
+    def _get_mail_url(self):
```

Comment:
By the way I propose to remove this indirection and use directly get_share_url in the send by email templates. That one other piece of code coming from old implementations that we can get rid of to clean a bit the code. There will be some update to do in website quote but it should not be difficult I think.